src/java.base/share/classes/javax/security/cert/Certificate.java

```diff
@@ -81,42 +82,30 @@ public Certificate() {}
      * @return true if the encoded forms of the two certificates
      *         match, false otherwise.
      */
+    @Override
     public boolean equals(Object other) {
         if (this == other)
             return true;
         if (!(other instanceof Certificate))
             return false;
         try {
-            byte[] thisCert = this.getEncoded();
-            byte[] otherCert = ((Certificate)other).getEncoded();
-
-            if (thisCert.length != otherCert.length)
-                return false;
-            for (int i = 0; i < thisCert.length; i++)
-                 if (thisCert[i] != otherCert[i])
-                     return false;
-            return true;
+            return Arrays.equals(this.getEncoded(),
+                    ((Certificate)other).getEncoded());
         } catch (CertificateException e) {
             return false;
         }
     }
 
     /**
-     * Returns a hashcode value for this certificate from its
-     * encoded form.
-     *
-     * @return the hashcode value.
+     * {@return a hashcode value for this certificate from its
+     * encoded form}
      */
+    @Override
     public int hashCode() {
-        int     retval = 0;
         try {
-            byte[] certData = this.getEncoded();
-            for (int i = 1; i < certData.length; i++) {
-                 retval += certData[i] * i;
-            }
-            return (retval);
+            return Arrays.hashCode(this.getEncoded());
         } catch (CertificateException e) {
```

Comment:
Ditto, what value at `certData[0]` is now included in the hash?

Reply:
Those are correct observations. Unspecified hashCode methods changed in this proposal might return different values than the ones they currently return in the mainline; hence my disclaimer in this PR description:

> This PR sacrifices compatibility for pragmatism: it changes some hashCode implementations to produce different values than before to allow more utilization of methods from Objects and Arrays. To my mind, those changes are **benign**. If you disagree, I'd be happy to discuss that and/or retract the concerning part of the change.

I'm waiting from someone from security-dev to tell me if that's okay to skip zeroth elements in both hashCode. It might've been accidental to skip it in the first place, or not. FWIW, their equals counterparts do not skip zeroth elements. On the other hand, javax.security.cert.Certificate is deprecated for removal, so I might as well exclude it.